### PR TITLE
uploading homework lesson 15

### PR DESCRIPTION
**Pull request by edisabou**

### CODE REVIEW

File: lesson_15/edmond_sabou/test_home1_les13.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import unittest
+
+
+class TestPoligons(unittest.TestCase):
+
+    def test_if_side_is_not_none(self):
+        sides = 1, "asd", 3
```
Comment (Liitle) [inline lesson_15/edmond_sabou/test_home1_les13.py]: the test is good, the only problem is that the code is not used in this test. You need to make an instance of Polygon with the sides you defined and make the asserts on the instance_of_poligon.sides

File: lesson_15/edmond_sabou/test_home1_les13.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+        self.assertIsNotNone(sides, "Value cannot be of type None")
+
+    def test_if_value_is_not_string(self):
+        x = 1
+        y = 2
```
Comment (Liitle) [inline lesson_15/edmond_sabou/test_home1_les13.py]: same as the above comment. This test is not tucking the code, the test will never fail

File: lesson_15/edmond_sabou/test_home1_les13.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+        tp_x = type(x)
```
Comment (Liitle) [inline lesson_15/edmond_sabou/test_home1_les13.py]: a better way of testing for type is to use `isinstance(x, int)`

**Review by Liitle — CHANGES_REQUESTED** [review]
The test cases are good, but you need to make sure you are testing the code. Create some instances and test on them

File: lesson_15/edmond_sabou/test_home1_les13.py (hunk 1 above)
Comment (edisabou) [reply lesson_15/edmond_sabou/test_home1_les13.py]: I had a feeling something was not ok. changed it now. please check

File: lesson_15/edmond_sabou/test_home1_les13.py (hunk 3 above)
Comment (edisabou) [reply lesson_15/edmond_sabou/test_home1_les13.py]: I had a feeling something was not ok. changed it now. please check

File: lesson_15/edmond_sabou/test_home1_les13.py (hunk 2 above)
Comment (edisabou) [reply lesson_15/edmond_sabou/test_home1_les13.py]: I had a feeling something was not ok. changed it now. please check

File: lesson_15/edmond_sabou/test_home1_les13.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import unittest
+
+from lesson_15.edmond_sabou import home1_les13
+
+
+class TestPoligons(unittest.TestCase):
+
+    def test_if_side_is_not_none(self):
+        sides = home1_les13.Polygon = 1
```
Comment (Liitle) [inline lesson_15/edmond_sabou/test_home1_les13.py]: you need an instance of Polygon. You used the class (home1_les13.Polygon) an instance will be `home1_les13.Polygon(1,1,1,1)`

Some for the rest of the examples use instances, not the class.

Another thing are the asserts. If you want to test that an object is not none. test with the proper assert: `self.assertIsNotNone()`.

**Review by Liitle — CHANGES_REQUESTED** [review]

File: lesson_15/edmond_sabou/test_home1_les13.py (hunk 4 above)
Comment (Liitle) [reply lesson_15/edmond_sabou/test_home1_les13.py]: @edisabou, can you please have another look at this PR and make the required changes

Comment (edisabou) [reply lesson_15/edmond_sabou/test_home1_les13.py]: instantiated classes. though could make `test_if_value_is_not_string` work. do you have any suggestions there please?

**Comment by Liitle** [comment]
You made the changes and now the `i` is an instance but you are still not testing on it. In PyCharm you will see a warning that tells you that `i` is never used, please make the asserts on `i` using the attributes you have on the instance. `i` has the attribute `sides`

**Comment by edisabou** [comment]
amended code and feeling really confident about this one.
though could not make the `test_if_value_is_not_string` function work, it's failing every time. can you help on that please?

**Comment by edisabou** [comment]
Hi guys,
Can you let me know if there are any more changes I need to make to the code?
Thanks.

**Comment by Liitle** [comment]
```
   def test_if_value_is_not_string(self):

        i = home1_les13.Polygon(1, 2, 2, 3)

        assert isinstance(i.sides, int)
```

`i.sides` is a tuple, not an integer. What you can do is test the tuple elements. 

```
   def test_if_value_is_not_string(self):

        i = home1_les13.Polygon(1, 2, 2, 3)
        
        for side in i.sides:
            assert isinstance(side, int)
```

you can also avoid the `for loop` by using `all()`

```
   def test_if_value_is_not_string(self):

        i = home1_les13.Polygon(1, 2, 2, 3)
      
        assert all([isinstance(side, int) for side in i.sides])

```

Your test is actually testing if `side` is an int, not that is not a string. To test that is not a string, you can do something like this

```
   def test_if_value_is_not_string(self):

        i = home1_les13.Polygon(1, 2, 2, 3)
      
        assert all([not isinstance(side, str) for side in i.sides])

```
Sorry for the late reply :smile:

**Comment by edisabou** [comment]
> ```
>    def test_if_value_is_not_string(self):
> 
>         i = home1_les13.Polygon(1, 2, 2, 3)
> 
>         assert isinstance(i.sides, int)
> ```
> 
> `i.sides` is a tuple, not an integer. What you can do is test the tuple elements.
> 
> ```
>    def test_if_value_is_not_string(self):
> 
>         i = home1_les13.Polygon(1, 2, 2, 3)
>         
>         for side in i.sides:
>             assert isinstance(side, int)
> ```
> 
> you can also avoid the `for loop` by using `all()`
> 
> ```
>    def test_if_value_is_not_string(self):
> 
>         i = home1_les13.Polygon(1, 2, 2, 3)
>       
>         assert all([isinstance(side, int) for side in i.sides])
> ```
> 
> Your test is actually testing if `side` is an int, not that is not a string. To test that is not a string, you can do something like this
> 
> ```
>    def test_if_value_is_not_string(self):
> 
>         i = home1_les13.Polygon(1, 2, 2, 3)
>       
>         assert all([not isinstance(side, str) for side in i.sides])
> ```
> 
> Sorry for the late reply smile

I've amended code. please let me know if there is anything else. thanks.

**Review by Liitle — APPROVED** [review]

**Pull request merged**